federation/pkg/federation-controller/service/dns.go

```diff
@@ -44,6 +44,10 @@ func (s *ServiceController) getHealthyEndpoints(clusterName string, cachedServic
 			return nil, nil, nil, err
 		}
 		for _, ingress := range lbStatus.Ingress {
+			_, ok := cachedService.endpointMap[lbClusterName]
```

Comment:
I added the checking, please review, thanks.

What I meant about later improvement is that this map could be changed to map[string]interface{}, so that we can set endpointMap[clusterName] = nil when ready address returns, and only rely on the key checking.